File: php/src/CodingInterview/Domain/PortfolioItem.php
<?php

declare(strict_types=1);

namespace Folio\CodingInterview\Domain;

/** ポートフォリオの銘柄ごとの構成比率を表す。 */
final class PortfolioItem
{
    public function __construct(
        public readonly StockSymbol $symbol,
        public readonly BigDecimal $rate,
    ) {
    }
}

File: java8/src/main/java/folio/codinginterview/domain/PortfolioItem.java
package folio.codinginterview.domain;

import java.math.BigDecimal;

/** ポートフォリオの銘柄ごとの構成比率を表す。 */
public final class PortfolioItem {
    private final StockSymbol symbol;
    private final BigDecimal rate;

    public PortfolioItem(StockSymbol symbol, BigDecimal rate) {
        this.symbol = symbol;
        this.rate = rate;
    }

    public StockSymbol symbol() { return symbol; }

    public BigDecimal rate() { return rate; }
}
